### PR TITLE
force cleanup for manually deleted worker and job pods

### LINKED ISSUE #1506: Worker resource cleanup if/when a worker pod manually deleted
When a worker pod is manually deleted, will all associated K8s resources be successfully cleaned up?  @krancour thinks there may be some hardening we can do here.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1506

### CODE REVIEW

File: v2/observer/jobs_test.go [hunk 1]
```diff
@@ -495,6 +495,39 @@ func TestDeleteJobResources(t *testing.T) {
...
+		},
+		syncMu: &sync.Mutex{},
+		jobsClient: &coreTesting.MockJobsClient{
+			UpdateStatusFn: func(
+				context.Context,
+				string,
+				string,
+				core.JobStatus,
+			) error {
+				return nil
```
Comment (vdice) [inline v2/observer/jobs_test.go:513]: Not sure if overkill... do we want to assert on expected status (ABORTED) here?

Similarly, do we want to add assertions on strings passed in to `CleanupFn`?  

(If either/both are added, these comments apply to the worker pod test as well.)

Comment (krancour) [reply v2/observer/jobs_test.go:513]: > Not sure if overkill.

Not sure either, but happy to add.

Comment (vdice) [reply v2/observer/jobs_test.go:513]: Ok, might as well add when convenient.  Thanks!

Comment (krancour) [reply v2/observer/jobs_test.go:514]: All set.

**Review by vdice — APPROVED** [review]
LGTM!

**Pull request merged**